addons/binding/org.openhab.binding.neeo/README.md

```diff
@@ -0,0 +1,254 @@
+
+
+## Channels
+
+### NEEO Brain
+
+The NEEO Brain has the following channels:
+| Channel Type ID    | Read/Write | Item Type    | Description                                                                                |
+|--------------------|------------|--------------|--------------------------------------------------------------------------------------------|
+| forwardActions     | R          | Trigger      | The forward actions channel                                                                |
+  
+
+The following properties are available at the time of this writing:
+
+| Name          | Description                                                                                                         |
+|---------------|---------------------------------------------------------------------------------------------------------------------|
+| AirKey        | Unknown (hints at a future airplay feature)                                                                         |
+| Version       | The software (not firmware) version of the NEEO Brain                                                               |
+| Is Configured | Whether the Brain has gone through it's initial setup (true) or not (false)                                         |
+| Label         | Internal label assigned to the Brain                                                                                |
+| Last Change   | The time (in milliseconds) that the Brain was last updated (Recipe/Devices/etc change - again not firmware)         |
+| Key           | The unique identifier of the Brain                                                                                  |
+| Name          | Internal name of the Brain                                                                                          |
+
+
+### NEEO Room
+
+The NEEO Room is dynamically generated from the Brain.  Each Room will dynamically generate the following channel groups:
+
+1) Each Room will have exactly one "room-state" representing the current state of the Room.
+2) Each Room will have zero or more "room-recipe-xxx" (where xxx is the Recipe key) groups representing each Recipe in the Room.
+3) Each Room will have zero or more "room-scenario-xxx" (where xxx is the Scenario key) groups representing each Scenario in the Room.
+
+#### Room State Group
+
+The following channels will be in the Room state group
+
+| Channel Type ID    | Read/Write | Item Type    | Description                                                                                |
+|--------------------|------------|--------------|--------------------------------------------------------------------------------------------|
+| activeScenarios    | R          | Trigger      | The name of the active Scenarios (JSON array of names.  Example: "['TV','DVR']")           |
```

Comment:
@martinvw asked for both of these channels to be triggers in a prior review.  I believe that the argument was that both this (and currentStep) are simply event notification from the brain and being an event notification - they should be trigger channels.

I fully agree with the 'currentStep' being a trigger because it really is just an event notification.  There is no state associated with it and it's even local to whatever process started the scenario (so if the remote starts the scenario - it get's the step notifications only, if openHAB starts the scenario, it get's the step notifications only).

I'm on the fence whether activeScenarios should be a trigger or item channel.  It's kinda a hybrid in that it's simply a notification from the brain - but depending on the underlying device, it may have state.  Example: starting a LIGHT, the scenario popup then goes away because lights don't have any state (so it's a notification).  However, start a TV - the activeScenario will include the TV until you are done watching it - so it has state.  What's your thoughts?  I'm leaning towards item channel for this one again just because it covers both situations.  This predated StringListType but I'll switch over to it with these changes.

Reply:
I see, so in case of the currentStep you only get a notification and you can query which step you are in, right? In this case I agree with it being a Trigger channel.

And if the activeScenarios cannot be queried in all cases either, it should stay a Trigger channel. However I would like it to have a user friendly payload, i.e. a StringListType which can be rendered by UIs instead of a pure JSON array. WDYT?

Reply:
Correct in both cases - I'll leave it as a trigger but change it to a StringListType.  

FYI - Off conversation topic but I'm also converting over to the new Channel*Builder(s) you committed as well (change will be in MetadataUtils.java) - so you'll see that as a change in the next commit.

Reply:
Actually - I think I see why I did a JSON string - trigger channel only accepts strings (so I can't give it a StringListType).  Maybe you see something I don't?

Reply:
Ok, let's take a step back: What is the purpose of this channel? Do you want to show the scenarios which are active in a UI? In this case this channel is clearly a state channel.

However, if you cannot fetch this information, i.e. react on a `REFRESH` command to obtain the current value for this channel, I have a hard time of thinking how this can be a useful information, i.e. if its not up2date most of the time.

The other option would be that you use the pushed information from the brain about the active scenarios for another purpose internally in the binding. So whenever the brain announces its active scenarios and the user wants to start one of them you just return instead of trying to start it again.

Reply:
I just looked up in the code that only `CallItem` supports the `StringListType`.

But did you see this part of my comment?

> However, if you cannot fetch this information, i.e. react on a REFRESH command to obtain the current value for this channel, I have a hard time of thinking how this can be a useful information, i.e. if its not up2date most of the time.

So I would rather vote for removing this channel.

Reply:
How about updating this channel with the information instead?

https://github.com/openhab/openhab2-addons/pull/3191/files/4d36ce3d7e6efd945b9556b8b4400e52ba9ec558#diff-1d97edbca8166155b4d0a32ae9c36b83R213

Reply:
You can fetch this information (that's why I was on the fence on this being a channel) and I know people do use this information (however, only in an event type of way - again, on the fence!).  However, you make a fair point with the status - let me think through some situations and see if they would ever be different (on the face, they shouldn't be different).  If they aren't, I'd tend to agree with you that maybe the channel is not needed.